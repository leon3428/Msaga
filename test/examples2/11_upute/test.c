int main(void) {
    int x = 0;
    int y = 0;

    int a = (int)'a';
    const char b = (const char)x;
    const int c = (const int)'a';
    char d = (char)((const int)300 + (int)'a');
    int e = (int)(char)(const int)(const char)(x + y);
    int f[10] = {1,2,3,4};
    int g = (char)f;

    return 0;
}