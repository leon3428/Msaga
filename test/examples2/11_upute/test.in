<prijevodna_jedinica>
 <vanjska_deklaracija>
  <definicija_funkcije>
   <ime_tipa>
    <specifikator_tipa>
     KR_INT 1 int
   IDN 1 main
   L_ZAGRADA 1 (
   KR_VOID 1 void
   D_ZAGRADA 1 )
   <slozena_naredba>
    L_VIT_ZAGRADA 1 {
    <lista_deklaracija>
     <lista_deklaracija>
      <lista_deklaracija>
       <lista_deklaracija>
        <lista_deklaracija>
         <lista_deklaracija>
          <lista_deklaracija>
           <lista_deklaracija>
            <lista_deklaracija>
             <deklaracija>
              <ime_tipa>
               <specifikator_tipa>
                KR_INT 2 int
              <lista_init_deklaratora>
               <init_deklarator>
                <izravni_deklarator>
                 IDN 2 x
                OP_PRIDRUZI 2 =
                <inicijalizator>
                 <izraz_pridruzivanja>
                  <log_ili_izraz>
                   <log_i_izraz>
                    <bin_ili_izraz>
                     <bin_xili_izraz>
                      <bin_i_izraz>
                       <jednakosni_izraz>
                        <odnosni_izraz>
                         <aditivni_izraz>
                          <multiplikativni_izraz>
                           <cast_izraz>
                            <unarni_izraz>
                             <postfiks_izraz>
                              <primarni_izraz>
                               BROJ 2 0
              TOCKAZAREZ 2 ;
            <deklaracija>
             <ime_tipa>
              <specifikator_tipa>
               KR_INT 3 int
             <lista_init_deklaratora>
              <init_deklarator>
               <izravni_deklarator>
                IDN 3 y
               OP_PRIDRUZI 3 =
               <inicijalizator>
                <izraz_pridruzivanja>
                 <log_ili_izraz>
                  <log_i_izraz>
                   <bin_ili_izraz>
                    <bin_xili_izraz>
                     <bin_i_izraz>
                      <jednakosni_izraz>
                       <odnosni_izraz>
                        <aditivni_izraz>
                         <multiplikativni_izraz>
                          <cast_izraz>
                           <unarni_izraz>
                            <postfiks_izraz>
                             <primarni_izraz>
                              BROJ 3 0
             TOCKAZAREZ 3 ;
           <deklaracija>
            <ime_tipa>
             <specifikator_tipa>
              KR_INT 5 int
            <lista_init_deklaratora>
             <init_deklarator>
              <izravni_deklarator>
               IDN 5 a
              OP_PRIDRUZI 5 =
              <inicijalizator>
               <izraz_pridruzivanja>
                <log_ili_izraz>
                 <log_i_izraz>
                  <bin_ili_izraz>
                   <bin_xili_izraz>
                    <bin_i_izraz>
                     <jednakosni_izraz>
                      <odnosni_izraz>
                       <aditivni_izraz>
                        <multiplikativni_izraz>
                         <cast_izraz>
                          L_ZAGRADA 5 (
                          <ime_tipa>
                           <specifikator_tipa>
                            KR_INT 5 int
                          D_ZAGRADA 5 )
                          <cast_izraz>
                           <unarni_izraz>
                            <postfiks_izraz>
                             <primarni_izraz>
                              ZNAK 5 'a'
            TOCKAZAREZ 5 ;
          <deklaracija>
           <ime_tipa>
            KR_CONST 6 const
            <specifikator_tipa>
             KR_CHAR 6 char
           <lista_init_deklaratora>
            <init_deklarator>
             <izravni_deklarator>
              IDN 6 b
             OP_PRIDRUZI 6 =
             <inicijalizator>
              <izraz_pridruzivanja>
               <log_ili_izraz>
                <log_i_izraz>
                 <bin_ili_izraz>
                  <bin_xili_izraz>
                   <bin_i_izraz>
                    <jednakosni_izraz>
                     <odnosni_izraz>
                      <aditivni_izraz>
                       <multiplikativni_izraz>
                        <cast_izraz>
                         L_ZAGRADA 6 (
                         <ime_tipa>
                          KR_CONST 6 const
                          <specifikator_tipa>
                           KR_CHAR 6 char
                         D_ZAGRADA 6 )
                         <cast_izraz>
                          <unarni_izraz>
                           <postfiks_izraz>
                            <primarni_izraz>
                             IDN 6 x
           TOCKAZAREZ 6 ;
         <deklaracija>
          <ime_tipa>
           KR_CONST 7 const
           <specifikator_tipa>
            KR_INT 7 int
          <lista_init_deklaratora>
           <init_deklarator>
            <izravni_deklarator>
             IDN 7 c
            OP_PRIDRUZI 7 =
            <inicijalizator>
             <izraz_pridruzivanja>
              <log_ili_izraz>
               <log_i_izraz>
                <bin_ili_izraz>
                 <bin_xili_izraz>
                  <bin_i_izraz>
                   <jednakosni_izraz>
                    <odnosni_izraz>
                     <aditivni_izraz>
                      <multiplikativni_izraz>
                       <cast_izraz>
                        L_ZAGRADA 7 (
                        <ime_tipa>
                         KR_CONST 7 const
                         <specifikator_tipa>
                          KR_INT 7 int
                        D_ZAGRADA 7 )
                        <cast_izraz>
                         <unarni_izraz>
                          <postfiks_izraz>
                           <primarni_izraz>
                            ZNAK 7 'a'
          TOCKAZAREZ 7 ;
        <deklaracija>
         <ime_tipa>
          <specifikator_tipa>
           KR_CHAR 8 char
         <lista_init_deklaratora>
          <init_deklarator>
           <izravni_deklarator>
            IDN 8 d
           OP_PRIDRUZI 8 =
           <inicijalizator>
            <izraz_pridruzivanja>
             <log_ili_izraz>
              <log_i_izraz>
               <bin_ili_izraz>
                <bin_xili_izraz>
                 <bin_i_izraz>
                  <jednakosni_izraz>
                   <odnosni_izraz>
                    <aditivni_izraz>
                     <multiplikativni_izraz>
                      <cast_izraz>
                       L_ZAGRADA 8 (
                       <ime_tipa>
                        <specifikator_tipa>
                         KR_CHAR 8 char
                       D_ZAGRADA 8 )
                       <cast_izraz>
                        <unarni_izraz>
                         <postfiks_izraz>
                          <primarni_izraz>
                           L_ZAGRADA 8 (
                           <izraz>
                            <izraz_pridruzivanja>
                             <log_ili_izraz>
                              <log_i_izraz>
                               <bin_ili_izraz>
                                <bin_xili_izraz>
                                 <bin_i_izraz>
                                  <jednakosni_izraz>
                                   <odnosni_izraz>
                                    <aditivni_izraz>
                                     <aditivni_izraz>
                                      <multiplikativni_izraz>
                                       <cast_izraz>
                                        L_ZAGRADA 8 (
                                        <ime_tipa>
                                         KR_CONST 8 const
                                         <specifikator_tipa>
                                          KR_INT 8 int
                                        D_ZAGRADA 8 )
                                        <cast_izraz>
                                         <unarni_izraz>
                                          <postfiks_izraz>
                                           <primarni_izraz>
                                            BROJ 8 300
                                     PLUS 8 +
                                     <multiplikativni_izraz>
                                      <cast_izraz>
                                       L_ZAGRADA 8 (
                                       <ime_tipa>
                                        <specifikator_tipa>
                                         KR_INT 8 int
                                       D_ZAGRADA 8 )
                                       <cast_izraz>
                                        <unarni_izraz>
                                         <postfiks_izraz>
                                          <primarni_izraz>
                                           ZNAK 8 'a'
                           D_ZAGRADA 8 )
         TOCKAZAREZ 8 ;
       <deklaracija>
        <ime_tipa>
         <specifikator_tipa>
          KR_INT 9 int
        <lista_init_deklaratora>
         <init_deklarator>
          <izravni_deklarator>
           IDN 9 e
          OP_PRIDRUZI 9 =
          <inicijalizator>
           <izraz_pridruzivanja>
            <log_ili_izraz>
             <log_i_izraz>
              <bin_ili_izraz>
               <bin_xili_izraz>
                <bin_i_izraz>
                 <jednakosni_izraz>
                  <odnosni_izraz>
                   <aditivni_izraz>
                    <multiplikativni_izraz>
                     <cast_izraz>
                      L_ZAGRADA 9 (
                      <ime_tipa>
                       <specifikator_tipa>
                        KR_INT 9 int
                      D_ZAGRADA 9 )
                      <cast_izraz>
                       L_ZAGRADA 9 (
                       <ime_tipa>
                        <specifikator_tipa>
                         KR_CHAR 9 char
                       D_ZAGRADA 9 )
                       <cast_izraz>
                        L_ZAGRADA 9 (
                        <ime_tipa>
                         KR_CONST 9 const
                         <specifikator_tipa>
                          KR_INT 9 int
                        D_ZAGRADA 9 )
                        <cast_izraz>
                         L_ZAGRADA 9 (
                         <ime_tipa>
                          KR_CONST 9 const
                          <specifikator_tipa>
                           KR_CHAR 9 char
                         D_ZAGRADA 9 )
                         <cast_izraz>
                          <unarni_izraz>
                           <postfiks_izraz>
                            <primarni_izraz>
                             L_ZAGRADA 9 (
                             <izraz>
                              <izraz_pridruzivanja>
                               <log_ili_izraz>
                                <log_i_izraz>
                                 <bin_ili_izraz>
                                  <bin_xili_izraz>
                                   <bin_i_izraz>
                                    <jednakosni_izraz>
                                     <odnosni_izraz>
                                      <aditivni_izraz>
                                       <aditivni_izraz>
                                        <multiplikativni_izraz>
                                         <cast_izraz>
                                          <unarni_izraz>
                                           <postfiks_izraz>
                                            <primarni_izraz>
                                             IDN 9 x
                                       PLUS 9 +
                                       <multiplikativni_izraz>
                                        <cast_izraz>
                                         <unarni_izraz>
                                          <postfiks_izraz>
                                           <primarni_izraz>
                                            IDN 9 y
                             D_ZAGRADA 9 )
        TOCKAZAREZ 9 ;
      <deklaracija>
       <ime_tipa>
        <specifikator_tipa>
         KR_INT 10 int
       <lista_init_deklaratora>
        <init_deklarator>
         <izravni_deklarator>
          IDN 10 f
          L_UGL_ZAGRADA 10 [
          BROJ 10 10
          D_UGL_ZAGRADA 10 ]
         OP_PRIDRUZI 10 =
         <inicijalizator>
          L_VIT_ZAGRADA 10 {
          <lista_izraza_pridruzivanja>
           <lista_izraza_pridruzivanja>
            <lista_izraza_pridruzivanja>
             <lista_izraza_pridruzivanja>
              <izraz_pridruzivanja>
               <log_ili_izraz>
                <log_i_izraz>
                 <bin_ili_izraz>
                  <bin_xili_izraz>
                   <bin_i_izraz>
                    <jednakosni_izraz>
                     <odnosni_izraz>
                      <aditivni_izraz>
                       <multiplikativni_izraz>
                        <cast_izraz>
                         <unarni_izraz>
                          <postfiks_izraz>
                           <primarni_izraz>
                            BROJ 10 1
             ZAREZ 10 ,
             <izraz_pridruzivanja>
              <log_ili_izraz>
               <log_i_izraz>
                <bin_ili_izraz>
                 <bin_xili_izraz>
                  <bin_i_izraz>
                   <jednakosni_izraz>
                    <odnosni_izraz>
                     <aditivni_izraz>
                      <multiplikativni_izraz>
                       <cast_izraz>
                        <unarni_izraz>
                         <postfiks_izraz>
                          <primarni_izraz>
                           BROJ 10 2
            ZAREZ 10 ,
            <izraz_pridruzivanja>
             <log_ili_izraz>
              <log_i_izraz>
               <bin_ili_izraz>
                <bin_xili_izraz>
                 <bin_i_izraz>
                  <jednakosni_izraz>
                   <odnosni_izraz>
                    <aditivni_izraz>
                     <multiplikativni_izraz>
                      <cast_izraz>
                       <unarni_izraz>
                        <postfiks_izraz>
                         <primarni_izraz>
                          BROJ 10 3
           ZAREZ 10 ,
           <izraz_pridruzivanja>
            <log_ili_izraz>
             <log_i_izraz>
              <bin_ili_izraz>
               <bin_xili_izraz>
                <bin_i_izraz>
                 <jednakosni_izraz>
                  <odnosni_izraz>
                   <aditivni_izraz>
                    <multiplikativni_izraz>
                     <cast_izraz>
                      <unarni_izraz>
                       <postfiks_izraz>
                        <primarni_izraz>
                         BROJ 10 4
          D_VIT_ZAGRADA 10 }
       TOCKAZAREZ 10 ;
     <deklaracija>
      <ime_tipa>
       <specifikator_tipa>
        KR_INT 11 int
      <lista_init_deklaratora>
       <init_deklarator>
        <izravni_deklarator>
         IDN 11 g
        OP_PRIDRUZI 11 =
        <inicijalizator>
         <izraz_pridruzivanja>
          <log_ili_izraz>
           <log_i_izraz>
            <bin_ili_izraz>
             <bin_xili_izraz>
              <bin_i_izraz>
               <jednakosni_izraz>
                <odnosni_izraz>
                 <aditivni_izraz>
                  <multiplikativni_izraz>
                   <cast_izraz>
                    L_ZAGRADA 11 (
                    <ime_tipa>
                     <specifikator_tipa>
                      KR_CHAR 11 char
                    D_ZAGRADA 11 )
                    <cast_izraz>
                     <unarni_izraz>
                      <postfiks_izraz>
                       <primarni_izraz>
                        IDN 11 f
      TOCKAZAREZ 11 ;
    <lista_naredbi>
     <naredba>
      <naredba_skoka>
       KR_RETURN 13 return
       <izraz>
        <izraz_pridruzivanja>
         <log_ili_izraz>
          <log_i_izraz>
           <bin_ili_izraz>
            <bin_xili_izraz>
             <bin_i_izraz>
              <jednakosni_izraz>
               <odnosni_izraz>
                <aditivni_izraz>
                 <multiplikativni_izraz>
                  <cast_izraz>
                   <unarni_izraz>
                    <postfiks_izraz>
                     <primarni_izraz>
                      BROJ 13 0
       TOCKAZAREZ 13 ;
    D_VIT_ZAGRADA 14 }
