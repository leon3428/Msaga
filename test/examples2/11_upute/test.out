<cast_izraz> ::= L_ZAGRADA(11,() <ime_tipa> D_ZAGRADA(11,)) <cast_izraz>
